swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -314,9 +316,11 @@ public final class MenuBar: NSObject, ObservableObject {
   }
 
   @objc private func quitButtonTapped() {
-    Task {
-      model.store.stop()
-      NSApp.terminate(self)
+    Task.detached { [weak self] in
+      guard let self else { return }
+
+      await self.model.store.stop()
+      await NSApp.terminate(self)
```

Comment:
How come these are suddenly `async` or why are they being awaited now and not before?

Reply:
That's a great question. It became a compiler error as soon as I moved these to `Task.detached`.

`Task.detached` doesn't run the closure in the same concurrency context as the caller (i.e. same thread) unlike regular `Task`.

Since these functions contain properties that live on the main thread (e.g. Store is annotated `@MainActor`) Swift forces you to use `await`.

<img width="768" alt="Screenshot 2025-01-16 at 6 14 41 AM" src="https://github.com/user-attachments/assets/fe4e30e7-f6ab-46d6-9f8c-6f27cc459f24" />